aspnetcore/blazor/forms-and-input-components.md

```diff
@@ -37,28 +37,90 @@ The <xref:Microsoft.AspNetCore.Components.Forms?displayProperty=fullName> namesp
 
 A project created from the Blazor project template includes the namespace by default in the app's `_Imports.razor` file, which makes the namespace available to the app's Razor components.
 
+:::moniker range=">= aspnetcore-8.0"
+
+## Enhanced navigation and form handling
+
+Examples adopt [enhanced navigation](xref:blazor/fundamentals/routing#enhanced-navigation-and-form-handling) for form POST requests with the `Enhance` parameter for `EditForm` forms or the `data-enhance` attribute for HTML forms (`<form>`):
+
+```razor
+<EditForm Enhance ...>
+    ...
+</EditForm>
+```
+
+```html
+<form data-enhance ...>
+    ...
+</form>
+```
+
+<span aria-hidden="true">❌</span><span class="visually-hidden">Unsupported:</span> You can't set enhanced navigation on a form's ancestor element to enable enhanced navigation for the form.
+
+```html
+<div data-enhance>
+    <form ...>
+        <!-- NOT enhanced -->
+    </form>
+</div>
+```
+
+> [!WARNING]
+> ***Don't try to enhance form navigation for forms that POST to non-Blazor endpoints.***
+>
+> This is the primary reason why enabling form POSTs with enhanced navigation requires an explicit gesture on each `EditForm` and `<form>` element. It isn't safe to assume that all forms in general should use enhanced posts because any given form might not target a Blazor endpoint.
```

Comment:
I think this warning might sound a bit too scary. I think all we really need to say is that enhanced forms only work with Blazor. Posting an enhanced form to non-Blazor endpoint will result in an error.